File: FrontEnd/lib/models/hotel_image.dart
class HotelImage {
  final int id;
  final int hotelId;
  final String imageUrl;
  final bool isThumbnail;

  HotelImage({
    required this.id,
    required this.hotelId,
    required this.imageUrl,
    required this.isThumbnail,
  });

  factory HotelImage.fromJson(Map<String, dynamic> json) {
    return HotelImage(
      id: json['id'],
      hotelId: json['hotel_id'],
      imageUrl: json['image_url'],
      isThumbnail: json['is_thumbnail'] == 1,
    );
  }
}

File: BackEnd/HotelBooking/src/main/java/com/example/HotelBoking/Entity/HotelImage.java
package com.example.HotelBoking.Entity;

import jakarta.persistence.*;

@Entity
@Table(name = "hotel_images")
public class HotelImage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "hotel_id", nullable = false)
    private Hotel hotel;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "is_thumbnail")
    private Boolean isThumbnail;

    public HotelImage() {}

    public HotelImage(Long id, Hotel hotel, String imageUrl, Boolean isThumbnail) {
        this.id = id;
        this.hotel = hotel;
        this.imageUrl = imageUrl;
        this.isThumbnail = isThumbnail;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Hotel getHotel() {
        return hotel;
    }
    public void setHotel(Hotel hotel) {
        this.hotel = hotel;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public Boolean getIsThumbnail() {
        return isThumbnail;
    }

    public void setIsThumbnail(Boolean isThumbnail) {
        this.isThumbnail = isThumbnail;
    }
}

